Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XldbDataStore.cs

```diff
@@ -68,14 +68,133 @@ public IEnumerable<string> GetEventsByType(ExecutionEventId eventTypeID)
+                                break;
+                            case ExecutionEventId.WorkerList:
+                                storedEvents.Add(WorkerListEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipExecutionPerformance:
+                                storedEvents.Add(PipExecutionPerformanceEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.DirectoryMembershipHashed:
+                                storedEvents.Add(DirectoryMembershipHashedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ProcessExecutionMonitoringReported:
+                                storedEvents.Add(ProcessExecutionMonitoringReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ProcessFingerprintComputation:
+                                storedEvents.Add(ProcessFingerprintComputationEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ExtraEventDataReported:
+                                storedEvents.Add(ExtraEventDataReported.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.DependencyViolationReported:
+                                storedEvents.Add(DependencyViolationReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipExecutionStepPerformanceReported:
+                                storedEvents.Add(PipExecutionStepPerformanceReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipCacheMiss:
+                                storedEvents.Add(PipCacheMissEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ResourceUsageReported:
+                                storedEvents.Add(StatusReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.DominoInvocation:
+                                storedEvents.Add(BXLInvocationEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipExecutionDirectoryOutputs:
+                                storedEvents.Add(PipExecutionDirectoryOutputsEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            default:
+                                break;

```

Comment:
Just for your consideration:
Usually we have an assert if we reach a default case to ensure we catch someone adding a new type and missing it.
Now here we are at an interesting point since we want to have a compatible version / layer :) So failing here goes directly against that :)

Perhaps it is worth to at a minimum to count these and include them in telemetry, or at least log it happened?

I'll leave it for your consideration. No need to hold off the PR for this though.

Reply:
I think it's valid point. I also think that we need to differentiate between "I looked for X and did not find it" and "I looked for X, found it, but I don't know who to deserialize it"

---
In reply to: [309475293](https://github.com/microsoft/BuildXL/pull/685#discussion_r309475293) [](ancestors = 309475293)